### PR TITLE
New changes

### PR DESCRIPTION
**Pull request by christianneurohr**
See git commit comments for details.

### CODE REVIEW

**Comment by pascalmolin** [comment]
Several general things first:
- now your changes are automatically tested, here you can see things fail
- do not keep your Makefile, I updated the main to handle all cases (me, you, travis)
- as you can see, I have put the relevant information for computing shifts in the edge_t, so that rho does not need to be computed

**Comment by pascalmolin** [comment]
Now concerning nth_root_turn : you can simplify. There is no need to multiply by -1 each time, just accumulate in an int and change sign at the end. And the signs can just be ints, not arb types.
Also you can remove blank lines at the end.

**Review by pascalmolin — CHANGES_REQUESTED** [review]
no tmp files! (here main.toc)

File: arb/intersections_tree.c [hunk 1]
```diff
@@ -35,46 +47,85 @@ fill_block(si_mat_t c, slong i, slong j, slong sp, slong sm, slong m)
...
+                }
             }
+            else if (ek.b == el.a)
+            {
+                /* case b=c  */
+                shft = round((1/(2*PI)) * ( PI + rho + el.va - ek.vb ));
+                sp = -shft;
+                sm = 1 - shft; 
+            }
+            else if (ek.a == el.b)
```
Comment (pascalmolin) [inline arb/intersections_tree.c:98]: this case does not occur

File: arb/intersections_tree.c [hunk 2]
```diff
@@ -35,46 +47,85 @@ fill_block(si_mat_t c, slong i, slong j, slong sp, slong sm, slong m)
...
 
-        /* intersection with others */
+        /* intersection with other cycles */
         for (l = k + 1; l < d - 1; l++)
         {
             edge_t el = tree->e[l];
 
-            if (el.a != ek.a && el.a != ek.b)
+            if (ek.a != el.a && ek.b != el.a)
                 /* no intersection */
```
Comment (pascalmolin) [inline arb/intersections_tree.c:67]: nice catch

**Comment by christianneurohr** [comment]
Thanks for the comments and updates. It were my first steps with arb, I
think it will be better soon!

2016-12-16 21:22 GMT+01:00 pascalmolin <notifications@github.com>:

**Comment by pascalmolin** [comment]
No problem. I will push the intersection tree file in a few minutes, I
included your changes and I will add a test file.
As you may have guessed from the code, I rewrote your theorem 5.1 in a more
compact way.

Then I think it will be easier to close the pull request so that you start
a new one with only a 2nd version of the nth_root.

An easy way to get started would also be to contribute tests, you see how
they are written. Some could check the output is the same as the one you
got from magma.
Do not hesitate to use branches to avoid messing up everything when a
commit is finally not taken.

Pascal

2016-12-16 21:55 GMT+01:00 christianneurohr <notifications@github.com>:

**Comment by christianneurohr** [comment]
Hello Pascal,

i pushed the reworked nth_root. I like your simplification of Thm 5.1., you
think it should be like that in the paper as well?
Also, your Makefile isn't working for me (cannot 'make' as it doesnt find
arb.h) and I don't know how to run the test files, any idea how to fix that?

Christian

2016-12-16 22:27 GMT+01:00 pascalmolin <notifications@github.com>:

**Pull request closed without merging**